### PR TITLE
Add generic client

### PR DESCRIPTION
**Pull request by fuchsnj**
makes `SqsClient` and `SqsConsumer` generic. The builder for `SqsClient` can accept functions for serializing, which will build the appropriate type of client.

### CODE REVIEW

File: src/main/java/com/bandwidth/sqs/client/SqsClientBuilder.java [hunk 1]
```diff
@@ -22,28 +24,53 @@
...
 
-    public SqsClient build() {
+
+    /**
+     * Converts the builder to a different type by defining mapping functions between the original and new type
+     *
+     * @param newDeserialize Map from the original value, to the new value
+     * @param newSerialize   Map from the new value, back to the original value
+     */
+    public <U> SqsClientBuilder<U> map(Function<T, U> newDeserialize, Function<U, T> newSerialize) {
```
Comment (derelbenkoenig) [inline src/main/java/com/bandwidth/sqs/client/SqsClientBuilder.java]: rename params

File: src/main/java/com/bandwidth/sqs/queue/SerializingSqsQueue.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import io.reactivex.Completable;
+import io.reactivex.Single;
+import io.reactivex.functions.Function;
+
+public class SerializingSqsQueue<A, B> implements SqsQueue<B> {
```
Comment (derelbenkoenig) [inline src/main/java/com/bandwidth/sqs/queue/SerializingSqsQueue.java]: by convention this would be `SerializingSqsQueue<T, U>`

Comment (derelbenkoenig) [inline src/main/java/com/bandwidth/sqs/queue/SerializingSqsQueue.java]: This is still a mapping that you're referring to as serialization

File: src/test/java/com/bandwidth/sqs/client/SqsClientBuilderTest.java [hunk 3]
```diff
@@ -16,11 +18,18 @@
 
     @Test
     public void testBuilder() {
-        SqsClient sqsClient = SqsClient.builder()
+        SqsClientBuilder<String> stringBuilder = SqsClient.builder()
```
Comment (derelbenkoenig) [inline src/test/java/com/bandwidth/sqs/client/SqsClientBuilderTest.java]: This isn't a `stringBuilder`, it's a `stringClientBuilder`

Comment (fuchsnj) [inline src/test/java/com/bandwidth/sqs/client/SqsClientBuilderTest.java]: the `Client` is silent

**Pull request merged**